chanbackup/pubsub.go

```diff
@@ -180,6 +196,44 @@ func (s *SubSwapper) Stop() error {
 	return nil
 }
 
+// ManualUpdate inserts/updates channel states into the swapper. The updates
+// are processed in another goroutine. If wait=true, the method waits for the
+// updates to be fully processed and the file updated on-disk before returning.
+func (s *SubSwapper) ManualUpdate(singles []Single, wait bool) error {
+	// Create the update object to insert into the processing loop.
+	errChan := make(chan error, 1)
+	update := manualUpdate{
+		singles: singles,
+		errChan: errChan,
+	}
+
+	select {
+	case s.manualUpdates <- update:
+	case <-s.quit:
+		return fmt.Errorf("swapper stopped when sending manual update")
+	}
+
+	if !wait {
+		// Don't wait for the processing.
+		return nil
+	}
+
+	// Wait for processing, block on errChan.
+	var err error
+	select {
+	case err = <-errChan:
```

Comment:
Handle the error directly in the case statement?

Reply:
Done :+1: